### ISSUE TITLE
FileInput - tab order and styling

### ISSUE THREAD

**Issue by jcfilben** [grommet/grommet#6372]
When navigating with the keyboard the tab order of FileInput is unexpected. First our focus is on the 'browse' button hitting tab again moves our focus to the main portion of the FileInput. This can be seen in the video below. It feels strange for focus to move right to left across the screen. Also the main portion of the FileInput gives us context for what the input field is so it should come first in the tab order before the browse button.

Additionally when not using a screenreader the main portion of the FileInput doesn't receive a focusIndicator. When navigating with the keyboard we should visually show a focus indicator on this section.

This issue is reproducable on the FileInput storybook stories: https://storybook.grommet.io/?path=/story/input-fileinput-max-size--max-size

Video of the issue with screenreader output:

https://user-images.githubusercontent.com/54560994/193368791-0fd2f60f-b1dd-4e55-9c74-4228dc80436a.mp4

**Comment by miqh** [comment]
@jcfilben,

I looked into this and I found it wasn't particularly difficult to correct both the keyboard navigation order and also provide a visual focus indicator on the actual file input element.

I'm happy to open up a pull request, but there's something bothering me.

I'm not an accessibility guru, but it seems redundant allowing navigation through both the file input as well as the browse anchor element. Hitting <kbd>ENTER</kbd> with either in focus essentially accomplishes the same thing (i.e. to spawn the system file picker dialog). That said, if you'd still like to see what was originally expected, I'll put those changes through.

![fileinput2](https://user-images.githubusercontent.com/43751307/193459353-d167956c-5086-41d7-8c1b-c00f15c63324.gif)

**Comment by jcfilben** [comment]
> I'm not an accessibility guru, but it seems redundant allowing navigation through both the file input as well as the browse anchor element. Hitting ENTER with either in focus essentially accomplishes the same thing (i.e. to spawn the system file picker dialog). That said, if you'd still like to see what was originally expected, I'll put those changes through.

Thanks for looking into this! This is a really good point. It does feel a bit redundant to be able to navigate to both places when the have the same action. Maybe we should not allow the browse anchor to receive keyboard focus. Does that feel like a reasonable approach?
